pkg/parquetconverter/converter_marker.go

```diff
@@ -0,0 +1,61 @@
+package parquetconverter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"path"
+
+	"github.com/cortexproject/cortex/pkg/storage/tsdb"
+	"github.com/efficientgo/core/errors"
+	"github.com/go-kit/log"
+	"github.com/oklog/ulid"
+	"github.com/thanos-io/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+const (
+	ParquetConverterMakerFileName = "parquet-converter-mark.json"
+	CurrentVersion                = 1
+)
+
+type CompactionMark struct {
+	Version int `json:"version"`
+}
+
+func (m *CompactionMark) markerFilename() string { return ParquetConverterMakerFileName }
+
+func ReadConverterMark(ctx context.Context, id ulid.ULID, userBkt objstore.InstrumentedBucket, logger log.Logger) (*CompactionMark, error) {
+	markerPath := path.Join(id.String(), ParquetConverterMakerFileName)
+	reader, err := userBkt.WithExpectedErrs(tsdb.IsOneOfTheExpectedErrors(userBkt.IsAccessDeniedErr, userBkt.IsObjNotFoundErr)).Get(ctx, markerPath)
+	if err != nil {
+		if userBkt.IsObjNotFoundErr(err) {
+			return &CompactionMark{}, nil
+		}
+
+		return &CompactionMark{}, err
+	}
+	defer runutil.CloseWithLogOnErr(logger, reader, "close bucket index reader")
+
+	metaContent, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, errors.Wrapf(err, "read file: %s", ParquetConverterMakerFileName)
+	}
+
+	marker := CompactionMark{}
+	err = json.Unmarshal(metaContent, &marker)
+	return &marker, err
+}
+
+func WriteCompactMark(ctx context.Context, id ulid.ULID, userBkt objstore.Bucket) error {
+	marker := CompactionMark{
+		Version: CurrentVersion,
+	}
+	markerPath := path.Join(id.String(), ParquetConverterMakerFileName)
```

Comment:
I wonder if we should also upload the marker into a common parquet marker path. This makes it easier to discover all files with parquet and I don't have to call `Exists` to check all blocks to see if they have parquet marker or not

Reply:
Done!